core/src/avm2.rs

```diff
@@ -213,15 +213,18 @@ impl<'gc> Avm2<'gc> {
 
     /// Dispatch an event on an object.
     ///
+    /// This should be a root activation. As such, the error is a message and not something actionable.
```

Comment:
This creates a "new local root" activation, but the comment that it "should" be a root sounds weird. This isn't a VM-wide root activation, as `a.addChild(b)` is enough to dispatch a new event from inside an existing activation.